File: src/types/talent.ts
export interface TalentData {
  id: string
  producerId?: string | null
  fullName: string
  email?: string | null
  phone?: string | null
  postalcode?: string | null
  street?: string | null
  neighborhood?: string | null
  city?: string | null
  numberAddress?: string | null
  complement?: string | null
  uf?: string | null
  document?: string | null
  birthDate?: Date | null
  age: number
  gender?: string | null
  inviteSent: boolean
  status: boolean
  dnaStatus: 'UNDEFINED' | 'COMPLETE' | 'PARTIAL'
  inviteSentAt?: Date | null
  inviteToken?: string | null
  clerkInviteId?: string | null
  updatedAt: Date
  createdAt: Date
  producer?: {
    id: string
    first_name: string
    last_name: string
    email: string
  } | null
  dna?: TalentDNAData | null
  files?: FileData[]
}

export interface TalentDNAData {
  id: string
  talentId: string
  height?: string | null
  weight?: string | null
  hairColor?: string | null
  hairType?: string | null
  hairLength?: string | null
  eyeColor?: string | null
  skinTone?: string | null
  chestSize?: string | null
  waistSize?: string | null
  hipSize?: string | null
  shoeSize?: string | null
  dressSize?: string | null
  pantsSize?: string | null
  shirtSize?: string | null
  jacketSize?: string | null
  faceShape?: string | null
  ethnicFeatures?: string | null
  bodyType?: string | null
  specialFeatures?: string | null
  accent?: string | null
  languages?: string | null
  intellectualDisability?: string | null
  physicalDisability?: string | null
  religion?: string | null
  travelAvailability: boolean
  visualDisability?: string | null
  createdAt: Date
  updatedAt: Date
}

export interface FileData {
  id: string
  url: string
  type: 'PHOTO' | 'VIDEO' | 'DOCUMENT' | 'COMPOSITE'
  talentId?: string | null
  projectId?: string | null
  uploadedAt: Date
  fileName?: string | null
  mimeType?: string | null
}

export interface ProducerData {
  id: string
  first_name: string
  last_name: string
  email: string
  code: string
}

export interface CreateTalentData {
  producerId?: string
  fullName: string
  email?: string
  phone?: string
  postalcode?: string
  street?: string
  neighborhood?: string
  city?: string
  numberAddress?: string
  complement?: string
  uf?: string
  document?: string
  birthDate?: Date
  age: number
  gender?: string
}

File: back/Pregiato.Core/Entities/Talent.cs
using System.ComponentModel.DataAnnotations;

namespace Pregiato.Core.Entities;

public class Talent
{
    public Guid Id { get; set; }
    
    [StringLength(50)]
    public string? ProducerId { get; set; }
    
    [Required]
    [StringLength(255)]
    public string FullName { get; set; } = string.Empty;
    
    [EmailAddress]
    [StringLength(255)]
    public string? Email { get; set; }
    
    [StringLength(20)]
    public string? Phone { get; set; }
    
    [StringLength(10)]
    public string? Postalcode { get; set; }
    
    [StringLength(255)]
    public string? Street { get; set; }
    
    [StringLength(100)]
    public string? Neighborhood { get; set; }
    
    [StringLength(100)]
    public string? City { get; set; }
    
    [StringLength(20)]
    public string? NumberAddress { get; set; }
    
    [StringLength(255)]
    public string? Complement { get; set; }
    
    [StringLength(2)]
    public string? Uf { get; set; }
    
    [StringLength(20)]
    public string? Document { get; set; }
    
    public DateTime? BirthDate { get; set; }
    
    [Required]
    [Range(0, 150)]
    public int Age { get; set; }
    
    [StringLength(20)]
    public string? Gender { get; set; }
    
    public bool InviteSent { get; set; } = false;
    
    public bool Status { get; set; } = true;
    
    [StringLength(20)]
    public string DnaStatus { get; set; } = "UNDEFINED";
    
    public DateTime? InviteSentAt { get; set; }
    
    [StringLength(255)]
    public string? InviteToken { get; set; }
    
    [StringLength(255)]
    public string? ClerkInviteId { get; set; }
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    
    // Navigation property
    public virtual TalentDNA? Dna { get; set; }
} 